mmv1/templates/terraform/post_create/resource_vertexai_endpoint_with_model_garden_deployment.go.tmpl

```diff
@@ -0,0 +1,59 @@
+log.Printf("[DEBUG] Beginning post_create for Vertex AI Endpoint with Model Garden Deployment")
+
+// Log Terraform resource data
+log.Printf("[DEBUG] Terraform Resource Data (d): ID=%s", d.Id())
+for key, val := range d.State().Attributes {
+	log.Printf("[DEBUG] d.State().Attributes[%s] = %s", key, val)
+}
+log.Printf("[DEBUG] d.Get(\"project\") = %v", d.Get("project"))
+log.Printf("[DEBUG] d.Get(\"location\") = %v", d.Get("location"))
+log.Printf("[DEBUG] d.Get(\"publisher_model_name\") = %v", d.Get("publisher_model_name"))
+
+// Log res structure
+log.Printf("[DEBUG] Top-level keys in res:")
+for k := range res {
+	log.Printf("[DEBUG]   - %s", k)
+}
+
+// Declare and populate opRes
+var opRes map[string]interface{}
+err = VertexAIOperationWaitTimeWithResponse(
```

Comment:
Are we not correctly following the operation in the normal create flow? Why do we need to handle it in a post_create?

Reply:
The created Endpoint resource is represented as a resource name in the LRO response. The post_create read the LRO response and read the Endpoint resource. This is the recommended approach from Terraform office hour.